src/mt.rs

```diff
@@ -67,20 +75,84 @@ impl Location {
     }
 }
 
+/// Configure a meta-tracer.
+pub struct MTBuilder {
+    hot_threshold: HotThreshold,
+}
+
+impl MTBuilder {
+    /// Create a meta-tracer with default parameters.
+    pub fn new() -> Self {
+        MTBuilder {
+            hot_threshold: DEFAULT_HOT_THRESHOLD,
+        }
+    }
+
+    /// Consume the `MTBuilder` and create a meta-tracer, returning the
+    /// [`MTThread`](struct.MTThread.html) representing the current thread.
+    pub fn init(self) -> MTThread {
+        MT::init(self.hot_threshold)
+    }
+
+    /// Change this meta-tracer builder's `hot_threshold` value.
+    pub fn hot_threshold(mut self, hot_threshold: HotThreshold) -> Self {
+        self.hot_threshold = hot_threshold;
+        self
+    }
+}
+
 /// A meta-tracer.
 pub struct MT {
     hot_threshold: HotThreshold,
+    active_threads: usize,
 }
 
 impl MT {
-    /// Create a new `MT` with default settings.
-    pub fn new() -> Self {
-        Self::new_with_hot_threshold(DEFAULT_HOT_THRESHOLD)
+    /// Create a new `MT`.
+    fn init(hot_threshold: HotThreshold) -> MTThread {
+        let mt = Arc::new(Mutex::new(Self {
+            hot_threshold,
+            active_threads: 1,
+        }));
+        MTThread::new(mt)
     }
 
-    /// Create a new `MT` with a specific hot threshold.
-    pub fn new_with_hot_threshold(hot_threshold: HotThreshold) -> Self {
-        Self { hot_threshold }
+    /// Create a new thread that can be used in the meta-tracer: the new thread that is created is
+    /// handed a [`MTThread`](struct.MTThread.html) from which the `MT` itself can be accessed.
+    pub fn spawn<F, T>(mt: Arc<Mutex<MT>>, f: F) -> io::Result<JoinHandle<T>>
+    where
+        F: FnOnce(MTThread) -> T,
+        F: Send + UnwindSafe + 'static,
+        T: Send + 'static,
+    {
+        let mt_cl = mt.clone();
+        thread::Builder::new().spawn(move || {
+            mt.lock().unwrap().active_threads += 1;
+            let r = catch_unwind(|| f(MTThread::new(mt_cl)));
+            mt.lock().unwrap().active_threads -= 1;
+            match r {
+                Ok(r) => r,
+                Err(e) => resume_unwind(e),
```

Comment:
The `resume_unwind` is simply because it's not our job to swallow errors: we do the operation we need to and carry on our merry way.

Reply:
Wait, I think I understand. We don't allow the thread to panic outright, because we need an opportunity to update the active thread count. So we catch any panic, decrement the count, then re-propogate the panic?

If that's the case, can we still add a comment?

Reply:
Am I correct?

Reply:
Correct. But I must admit, I think the code is simple and clear in this regard. A comment feels like overkill to me.

Reply:
It's up to you, but It wasn't immediately clear to me.

Reply:
I think I'll forego a comment here.